src/Service.Host/client/src/components/StockPools.js

```diff
@@ -0,0 +1,309 @@
+            const currentRow = stockPools?.find(r => r.id === searchDialogOpen.forRow);
+            let newRow = {
+                ...currentRow,
+                updated: true
+            };
+            c.searchUpdateFieldNames?.forEach(f => {
+                newRow = { ...newRow, [f.fieldName]: selected[f.searchResultFieldName] };
+            });
+
+            processRowUpdate(newRow, currentRow);
+            setSearchDialogOpen({ forRow: null, forColumn: null });
+        };
+
+        return (
+            <div id={c.field}>
+                <Dialog open={searchDialogOpen.forColumn === c.field} onClose={handleClose}>
+                    <DialogTitle>Search</DialogTitle>
+                    <DialogContent>
+                        <Search
+                            autoFocus
+                            propertyName={`${c.field}-searchTerm`}
+                            label="defaultLocation"
+                            resultsInModal
+                            resultLimit={100}
+                            value={searchTerm}
+                            loading={storageLocationsLoading}
+                            handleValueChange={(_, newVal) => setSearchTerm(newVal)}
+                            search={() => getStorageLocations(null, `?searchTerm=${searchTerm}`)}
+                            searchResults={storageLocationsResult?.map(s => ({
+                                ...s,
+                                id: s.name
+                            }))}
+                            priorityFunction="closestMatchesFirst"
+                            onResultSelect={handleSearchResultSelect}
+                            clearSearch={() => {}}
+                        />
+                    </DialogContent>
+                    <DialogActions>
+                        <Button onClick={handleClose}>Close</Button>
+                    </DialogActions>
+                </Dialog>
+            </div>
+        );
+    };
+
+    const stockPoolColumns = [
+        {
+            field: 'stockPoolCode',
+            headerName: 'Code',
+            editable: params => params.row?.creating === true,
+            width: 100
+        },
+        { field: 'stockPoolDescription', editable: true, headerName: 'Description', width: 225 },
+        {
+            field: 'dateInvalid',
+            headerName: 'Date Invalid',
+            editable: true,
+            width: 175,
+            valueFormatter: value => value && moment(value).format('DD-MMM-YYYY')
+        },
+        { field: 'stockCategory', editable: true, headerName: 'Stock Category', width: 100 },
+        {
+            field: 'accountingCompanyCode',
+            editable: true,
+            headerName: 'Accounting Company',
+            width: 100,
+            defaultFormat: true,
+            valueOptions: accountingCompany?.map(ac => ac.name),
+            type: 'singleSelect'
+        },
+        {
+            field: 'defaultLocation',
```

Comment:
this will just display the location id i think which people will not understand. Would need to display the name (while sending the id to the back end) or perhaps show both